File: app/src/main/java/com/wegielek/signalychinese/viewmodels/HistoryViewModel.java
package com.wegielek.signalychinese.viewmodels;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.wegielek.signalychinese.database.SearchHistory;
import com.wegielek.signalychinese.repository.DictionaryRepository;

import java.util.List;

public class HistoryViewModel extends AndroidViewModel {

    private final DictionaryRepository mDictionaryRepository;

    public HistoryViewModel(@NonNull Application application) {
        super(application);
        mDictionaryRepository = new DictionaryRepository(application);
    }

    public LiveData<List<SearchHistory>> getWholeHistory() {
        return mDictionaryRepository.getWholeHistory();
    }


}

File: app/src/main/java/com/wegielek/signalychinese/viewmodels/HistoryViewModel.kt
package com.wegielek.signalychinese.viewmodels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.google.common.util.concurrent.ListenableFuture
import com.wegielek.signalychinese.database.History
import com.wegielek.signalychinese.repository.DictionaryRepository

class HistoryViewModel(application: Application) : AndroidViewModel(application) {
    private val mDictionaryRepository: DictionaryRepository = DictionaryRepository(application)

    val wholeHistory: LiveData<List<History>>
        get() = mDictionaryRepository.wholeHistory

    fun deleteWholeHistory(): ListenableFuture<Void?> {
        return mDictionaryRepository.deleteWholeHistory()
    }
}
